### PR TITLE
Lint nested ++

### PR DESCRIPTION
**Pull request by sandersn**
I took Wesley's PR #5878 and fixed the lint. There are about 64 places that change. I also tightened the rule back to to disallow ++ in element access expressions.

### CODE REVIEW

File: scripts/tslint/noIncrementDecrementRule.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import * as Lint from "tslint/lib/lint";
+import * as ts from "typescript";
+
+
+export class Rule extends Lint.Rules.AbstractRule {
+    public static POSTFIX_FAILURE_STRING = "Don't use '++' or '--' postfix operators outside statements, for loops, or element access expressions.";
```
Comment (weswigham) [inline scripts/tslint/noIncrementDecrementRule.ts]: You need to fix the error message to no longer claim element access expressions are an exception.

File: src/compiler/utilities.ts [hunk 2]
```diff
@@ -1884,8 +1884,8 @@ namespace ts {
             writeTextOfNode,
             writeLiteral,
             writeLine,
-            increaseIndent: () => indent++,
-            decreaseIndent: () => indent--,
+            increaseIndent: () => { indent++; },
```
Comment (weswigham) [inline src/compiler/utilities.ts]: This is a meaningful change - these functions no longer return the indent level. Does this represent a shortcoming in the rule, and should they be reverted?

Comment (DanielRosenwasser) [inline src/compiler/utilities.ts]: No, I think the rule itself is effectively that you have to separate the mutation of your variable from reading its (potentially new) value. If people find this too restrictive, we can always change it back.

**Comment by DanielRosenwasser** [comment]
:+1:

File: src/compiler/utilities.ts (hunk 2 above)
Comment (weswigham) [inline src/compiler/utilities.ts]: If that's the case, should these functions be rewritten as `increaseIndent: () => { indent++; return indent; }` to preserve the old behavior? Or does it not matter?

Comment (sandersn) [inline src/compiler/utilities.ts]: It doesn't matter -- `EmitTextWriter` says these functions are void, so callers cannot have used the return value.

(Yes, we are saner than we look!)

File: scripts/tslint/noIncrementDecrementRule.ts (hunk 1 above)
Comment (sandersn) [inline scripts/tslint/noIncrementDecrementRule.ts]: done

**Pull request merged**